> how weird that I didn't hit this...

Heads-up @vorburger, About an hour or so ago (*probably before you wrote your above comment*) I pulled down [*your java11 branch*](http://bit.ly/VrbrgrJ11) to see what I could make of [*the apache#609 failure*](http://bit.ly/Trvs609J11) you mentioned in our earlier conversation. 

The very first thing I always do immediately after pulling down a new branch is I run *`gradle check`* to make sure all is well. This is what I got on your *`java11`* branch:

```
FAILURE: Build failed with an exception.
...
* What went wrong:
A problem occurred evaluating root project 'fineract-provider'.
> Could not get unknown property 'classesDir' for main classes of type org.gradle.api.internal.tasks.DefaultSourceSetOutput.
...
BUILD FAILED in 2m 19s
```

A grep on *`classesDir`* turned up this:

```
$ grep -n -R 'classesDir' fineract-provider
fineract-provider/build.gradle:75:    files = fileTree(sourceSets.main.output.classesDir).matching {
fineract-provider/build.gradle:329:sourceSets.main.output.resourcesDir = sourceSets.main.output.classesDir
fineract-provider/build.gradle:330:sourceSets.test.output.resourcesDir = sourceSets.test.output.classesDir
fineract-provider/build.gradle:369:    testClassesDir = project.sourceSets.integrationTest.output.classesDir

```

--- a/fineract-provider/build.gradle
+++ b/fineract-provider/build.gradle
@@ -187,6 +187,7 @@ configurations {
 	}
 	runtime
 	all*.exclude group: 'commons-logging'
+	all*.exclude module: 'jul-to-slf4j' // see FINERACT-700 re. ClassCircularityError: java/util/logging/LogRecord
 }
 /* Pick up dependencies based on the environemnt, defaults to production */
 if (project.hasProperty('env') && project.getProperty('env') == 'dev') {

--- a/fineract-provider/build.gradle
+++ b/fineract-provider/build.gradle
@@ -333,6 +333,23 @@ tomcatRunWar {
     configFile = file('src/test/resources/META-INF/context.xml')
 }
 
+task tomcatRunWarAsDaemonForIntegrationTest(type: com.bmuschko.gradle.tomcat.tasks.TomcatRunWar) {
+    httpPort = 8080
+    httpsPort = 8443
+    stopPort = 8081
+    stopKey=   'stopKey'
+    enableSSL = true
+    keystoreFile = file('src/main/resources/keystore.jks')
+    keystorePass = 'openmf'
+    configFile = file('src/test/resources/META-INF/context.xml')
+    daemon = true
+}
+
+task tomcatStopAfterIntegrationTest(type: com.bmuschko.gradle.tomcat.tasks.TomcatStop) {
+    stopPort = 8081
+    stopKey = 'stopKey'
+}
+
 /* http://stackoverflow.com/questions/19653311/jpa-repository-works-in-idea-and-production-but-not-in-gradle */
 sourceSets.main.output.resourcesDir = sourceSets.main.java.outputDir
 sourceSets.test.output.resourcesDir = sourceSets.test.java.outputDir
@@ -368,11 +385,12 @@ configurations {
 }
 
 task integrationTest(type:Test) {
-    description = "Run integration tests (located in src/integrationTest/java). Starts tomcat in daemon mode before executing the tests."
-    it.dependsOn war
-    doFirst {
-        tomcatRunWar.daemon = true
-    }
+    description = "Run integration tests (located in src/integrationTest/java). Starts Tomcat in daemon mode before executing the tests (and stops it after)."
+    dependsOn war
+    dependsOn tomcatRunWarAsDaemonForIntegrationTest
+    tasks.findByName('tomcatRunWarAsDaemonForIntegrationTest').mustRunAfter 'war'
+    finalizedBy tomcatStopAfterIntegrationTest
+
     testClassesDirs = project.sourceSets.integrationTest.output.classesDirs
     classpath = project.sourceSets.integrationTest.runtimeClasspath
 }

